site/app/controllers/HomePageController.php

```diff
@@ -41,6 +42,53 @@ public function showLibrary() {
     }
 
     /**
+     * @Route("/homework/library/search", methods={"GET"})
+     * @return Response
+     */
+    public function searchLibrary($path = "/usr/local/submitty/library") {
+        $gradeable_ids = FileUtils::getAllDirs($path);
```

Comment:
What if somebody uploads a zip file with a folder in the top level. You'll only see the top level folder and none of the gradeables insdie. You should look for the config.json file. This applies to all of your search functions.

Reply:
This function now returns routes for all config.json files found within the library folder.